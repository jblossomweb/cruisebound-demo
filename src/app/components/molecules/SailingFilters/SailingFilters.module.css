.filters {
}

.filter {
  @apply pb-4;
}
