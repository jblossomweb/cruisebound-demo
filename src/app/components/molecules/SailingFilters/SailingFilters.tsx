import React from 'react'

import type SelectOption from 'app/types/SelectOption'
import Dropdown from 'app/components/atoms/Dropdown'

import styles from './SailingFilters.module.css'

export interface SailingFiltersProps {
  portOptions: SelectOption[]
  cruiselineOptions: SelectOption[]
  selectedPort?: SelectOption['value']
  selectedCruiseline?: SelectOption['value']
  // eslint-disable-next-line no-unused-vars
  onPortChange: (port: string) => void
  // eslint-disable-next-line no-unused-vars
  onCruiselineChange: (line: string) => void
}

const SailingFilters: React.FC<SailingFiltersProps> = ({
  portOptions,
  cruiselineOptions,
  selectedPort = undefined,
  selectedCruiseline = undefined,
  onPortChange,
  onCruiselineChange,
}) => (
  <div className={styles.filters}>
    <div className={styles.filter}>
      <Dropdown
        label="Departure port"
        options={portOptions}
        value={selectedPort}
        onSelect={onPortChange}
        wide
        naked
      />
    </div>
    <div className={styles.filter}>
      <Dropdown
        label="Cruiseline"
        options={cruiselineOptions}
        value={selectedCruiseline}
        onSelect={onCruiselineChange}
        wide
        naked
      />
    </div>
  </div>
)

export default SailingFilters
