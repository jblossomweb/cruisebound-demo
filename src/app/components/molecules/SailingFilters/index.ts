export type { SailingFiltersProps } from './SailingFilters'
export { default } from './SailingFilters'
