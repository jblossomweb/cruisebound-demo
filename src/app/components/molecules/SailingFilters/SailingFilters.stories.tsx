import React from 'react'
import type { StoryFn, Meta } from '@storybook/react'

import type Sailing from 'app/types/Sailing'
import type SelectOption from 'app/types/SelectOption'
import mockResponse from 'test/__mocks__/sailings_response.json'
import getUniqueCruiselines from 'app/utils/getUniqueCruiselines'
import getUniqueDeparturePorts from 'app/utils/getUniqueDeparturePorts'
import SailingFilters, { type SailingFiltersProps } from './SailingFilters'

const mockSailings = mockResponse.results as Sailing[]
const allCruiselines = getUniqueCruiselines(mockSailings)
const allDeparturePorts = getUniqueDeparturePorts(mockSailings)
const makeOptions = (items: string[]): SelectOption[] => [
  {
    display: 'Any',
    value: 'any',
  },
  ...items.map((item) => ({
    display: item,
    value: item,
  })),
]

export default {
  title: 'components/molecules/SailingFilters',
  component: SailingFilters,
  argTypes: {
    portOptions: { control: 'object' },
    cruiselineOptions: { control: 'object' },
    selectedPort: { control: 'text' },
    selectedCruiseline: { control: 'text' },
    onPortChange: { control: 'none' },
    onCruiselineChange: { control: 'none' },
  },
} as Meta<typeof SailingFilters>

const Template: StoryFn<typeof SailingFilters> = (
  args: SailingFiltersProps
) => <SailingFilters {...args} />

export const Sample = Template.bind({})
Sample.args = {
  portOptions: makeOptions(allDeparturePorts),
  cruiselineOptions: makeOptions(allCruiselines),
  onPortChange: () => {
    // void
  },
  onCruiselineChange: () => {
    // void
  },
}

export const SelectedPort = Template.bind({})
SelectedPort.args = {
  ...Sample.args,
  selectedPort: allDeparturePorts[Math.floor(allDeparturePorts.length / 2)],
}

export const SelectedCruiseline = Template.bind({})
SelectedCruiseline.args = {
  ...Sample.args,
  selectedCruiseline: allCruiselines[Math.floor(allCruiselines.length / 2)],
}

export const SelectedBoth = Template.bind({})
SelectedBoth.args = {
  ...SelectedPort.args,
  ...SelectedCruiseline.args,
}

export const NoOptions = Template.bind({})
NoOptions.args = {
  ...Sample.args,
  portOptions: [],
  cruiselineOptions: [],
}
