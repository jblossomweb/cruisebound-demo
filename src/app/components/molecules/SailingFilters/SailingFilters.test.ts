import storySnaps from 'test/utils/storySnaps'
import * as stories from './SailingFilters.stories'

describe('app/components/molecules/SailingFilters', () => {
  describe('stories', () => {
    storySnaps(stories)
  })
})
