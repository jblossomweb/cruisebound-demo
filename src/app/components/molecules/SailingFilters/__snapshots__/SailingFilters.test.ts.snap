// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/molecules/SailingFilters stories NoOptions matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="filters"
  >
    <div
      class="filter"
    >
      <label
        class="label"
      >
        Departure port
      </label>
      <div
        class="select wide naked"
      >
        <select
          class="inner"
          data-testid="select"
        />
      </div>
    </div>
    <div
      class="filter"
    >
      <label
        class="label"
      >
        Cruiseline
      </label>
      <div
        class="select wide naked"
      >
        <select
          class="inner"
          data-testid="select"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`app/components/molecules/SailingFilters stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="filters"
  >
    <div
      class="filter"
    >
      <label
        class="label"
      >
        Departure port
      </label>
      <div
        class="select wide naked"
      >
        <select
          class="inner"
          data-testid="select"
        >
          <option
            value="any"
          >
            Any
          </option>
          <option
            value="Naples"
          >
            Naples
          </option>
          <option
            value="Sydney"
          >
            Sydney
          </option>
          <option
            value="Brisbane"
          >
            Brisbane
          </option>
          <option
            value="Singapore"
          >
            Singapore
          </option>
          <option
            value="Le Havre (Paris)"
          >
            Le Havre (Paris)
          </option>
          <option
            value="Colon"
          >
            Colon
          </option>
          <option
            value="Barcelona"
          >
            Barcelona
          </option>
          <option
            value="Port Canaveral"
          >
            Port Canaveral
          </option>
          <option
            value="Los Angeles"
          >
            Los Angeles
          </option>
          <option
            value="Fort Lauderdale (Port Everglades)"
          >
            Fort Lauderdale (Port Everglades)
          </option>
          <option
            value="Sockhna"
          >
            Sockhna
          </option>
          <option
            value="Lisbon"
          >
            Lisbon
          </option>
          <option
            value="Miami"
          >
            Miami
          </option>
          <option
            value="Isla Baltra"
          >
            Isla Baltra
          </option>
          <option
            value="Genoa"
          >
            Genoa
          </option>
          <option
            value="Hamburg"
          >
            Hamburg
          </option>
          <option
            value="Palermo"
          >
            Palermo
          </option>
          <option
            value="La Spezia (Cinque Terre)"
          >
            La Spezia (Cinque Terre)
          </option>
          <option
            value="San Francisco"
          >
            San Francisco
          </option>
          <option
            value="New York"
          >
            New York
          </option>
          <option
            value="Honolulu"
          >
            Honolulu
          </option>
          <option
            value="San Diego"
          >
            San Diego
          </option>
          <option
            value="Abu Dhabi"
          >
            Abu Dhabi
          </option>
          <option
            value="New Orleans"
          >
            New Orleans
          </option>
          <option
            value="Cape Town"
          >
            Cape Town
          </option>
          <option
            value="Jeddah"
          >
            Jeddah
          </option>
          <option
            value="Rotterdam"
          >
            Rotterdam
          </option>
          <option
            value="Tenerife"
          >
            Tenerife
          </option>
          <option
            value="Civitavecchia (Rome)"
          >
            Civitavecchia (Rome)
          </option>
          <option
            value="Mosquera Islet"
          >
            Mosquera Islet
          </option>
          <option
            value="Dubai"
          >
            Dubai
          </option>
          <option
            value="Tokyo"
          >
            Tokyo
          </option>
          <option
            value="Palma de Mallorca"
          >
            Palma de Mallorca
          </option>
          <option
            value="Southampton"
          >
            Southampton
          </option>
        </select>
      </div>
    </div>
    <div
      class="filter"
    >
      <label
        class="label"
      >
        Cruiseline
      </label>
      <div
        class="select wide naked"
      >
        <select
          class="inner"
          data-testid="select"
        >
          <option
            value="any"
          >
            Any
          </option>
          <option
            value="MSC Cruises"
          >
            MSC Cruises
          </option>
          <option
            value="Holland America Line"
          >
            Holland America Line
          </option>
          <option
            value="Princess Cruises"
          >
            Princess Cruises
          </option>
          <option
            value="Royal Caribbean International"
          >
            Royal Caribbean International
          </option>
          <option
            value="Norwegian Cruise Line"
          >
            Norwegian Cruise Line
          </option>
          <option
            value="Seabourn Cruise Line"
          >
            Seabourn Cruise Line
          </option>
          <option
            value="Carnival Cruise Line"
          >
            Carnival Cruise Line
          </option>
          <option
            value="Celebrity Cruises"
          >
            Celebrity Cruises
          </option>
          <option
            value="Oceania Cruises"
          >
            Oceania Cruises
          </option>
          <option
            value="Azamara"
          >
            Azamara
          </option>
          <option
            value="Cunard"
          >
            Cunard
          </option>
        </select>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`app/components/molecules/SailingFilters stories SelectedBoth matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="filters"
  >
    <div
      class="filter"
    >
      <label
        class="label"
      >
        Departure port
      </label>
      <div
        class="select wide naked"
      >
        <select
          class="inner"
          data-testid="select"
        >
          <option
            value="any"
          >
            Any
          </option>
          <option
            value="Naples"
          >
            Naples
          </option>
          <option
            value="Sydney"
          >
            Sydney
          </option>
          <option
            value="Brisbane"
          >
            Brisbane
          </option>
          <option
            value="Singapore"
          >
            Singapore
          </option>
          <option
            value="Le Havre (Paris)"
          >
            Le Havre (Paris)
          </option>
          <option
            value="Colon"
          >
            Colon
          </option>
          <option
            value="Barcelona"
          >
            Barcelona
          </option>
          <option
            value="Port Canaveral"
          >
            Port Canaveral
          </option>
          <option
            value="Los Angeles"
          >
            Los Angeles
          </option>
          <option
            value="Fort Lauderdale (Port Everglades)"
          >
            Fort Lauderdale (Port Everglades)
          </option>
          <option
            value="Sockhna"
          >
            Sockhna
          </option>
          <option
            value="Lisbon"
          >
            Lisbon
          </option>
          <option
            value="Miami"
          >
            Miami
          </option>
          <option
            value="Isla Baltra"
          >
            Isla Baltra
          </option>
          <option
            value="Genoa"
          >
            Genoa
          </option>
          <option
            value="Hamburg"
          >
            Hamburg
          </option>
          <option
            value="Palermo"
          >
            Palermo
          </option>
          <option
            value="La Spezia (Cinque Terre)"
          >
            La Spezia (Cinque Terre)
          </option>
          <option
            value="San Francisco"
          >
            San Francisco
          </option>
          <option
            value="New York"
          >
            New York
          </option>
          <option
            value="Honolulu"
          >
            Honolulu
          </option>
          <option
            value="San Diego"
          >
            San Diego
          </option>
          <option
            value="Abu Dhabi"
          >
            Abu Dhabi
          </option>
          <option
            value="New Orleans"
          >
            New Orleans
          </option>
          <option
            value="Cape Town"
          >
            Cape Town
          </option>
          <option
            value="Jeddah"
          >
            Jeddah
          </option>
          <option
            value="Rotterdam"
          >
            Rotterdam
          </option>
          <option
            value="Tenerife"
          >
            Tenerife
          </option>
          <option
            value="Civitavecchia (Rome)"
          >
            Civitavecchia (Rome)
          </option>
          <option
            value="Mosquera Islet"
          >
            Mosquera Islet
          </option>
          <option
            value="Dubai"
          >
            Dubai
          </option>
          <option
            value="Tokyo"
          >
            Tokyo
          </option>
          <option
            value="Palma de Mallorca"
          >
            Palma de Mallorca
          </option>
          <option
            value="Southampton"
          >
            Southampton
          </option>
        </select>
      </div>
    </div>
    <div
      class="filter"
    >
      <label
        class="label"
      >
        Cruiseline
      </label>
      <div
        class="select wide naked"
      >
        <select
          class="inner"
          data-testid="select"
        >
          <option
            value="any"
          >
            Any
          </option>
          <option
            value="MSC Cruises"
          >
            MSC Cruises
          </option>
          <option
            value="Holland America Line"
          >
            Holland America Line
          </option>
          <option
            value="Princess Cruises"
          >
            Princess Cruises
          </option>
          <option
            value="Royal Caribbean International"
          >
            Royal Caribbean International
          </option>
          <option
            value="Norwegian Cruise Line"
          >
            Norwegian Cruise Line
          </option>
          <option
            value="Seabourn Cruise Line"
          >
            Seabourn Cruise Line
          </option>
          <option
            value="Carnival Cruise Line"
          >
            Carnival Cruise Line
          </option>
          <option
            value="Celebrity Cruises"
          >
            Celebrity Cruises
          </option>
          <option
            value="Oceania Cruises"
          >
            Oceania Cruises
          </option>
          <option
            value="Azamara"
          >
            Azamara
          </option>
          <option
            value="Cunard"
          >
            Cunard
          </option>
        </select>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`app/components/molecules/SailingFilters stories SelectedCruiseline matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="filters"
  >
    <div
      class="filter"
    >
      <label
        class="label"
      >
        Departure port
      </label>
      <div
        class="select wide naked"
      >
        <select
          class="inner"
          data-testid="select"
        >
          <option
            value="any"
          >
            Any
          </option>
          <option
            value="Naples"
          >
            Naples
          </option>
          <option
            value="Sydney"
          >
            Sydney
          </option>
          <option
            value="Brisbane"
          >
            Brisbane
          </option>
          <option
            value="Singapore"
          >
            Singapore
          </option>
          <option
            value="Le Havre (Paris)"
          >
            Le Havre (Paris)
          </option>
          <option
            value="Colon"
          >
            Colon
          </option>
          <option
            value="Barcelona"
          >
            Barcelona
          </option>
          <option
            value="Port Canaveral"
          >
            Port Canaveral
          </option>
          <option
            value="Los Angeles"
          >
            Los Angeles
          </option>
          <option
            value="Fort Lauderdale (Port Everglades)"
          >
            Fort Lauderdale (Port Everglades)
          </option>
          <option
            value="Sockhna"
          >
            Sockhna
          </option>
          <option
            value="Lisbon"
          >
            Lisbon
          </option>
          <option
            value="Miami"
          >
            Miami
          </option>
          <option
            value="Isla Baltra"
          >
            Isla Baltra
          </option>
          <option
            value="Genoa"
          >
            Genoa
          </option>
          <option
            value="Hamburg"
          >
            Hamburg
          </option>
          <option
            value="Palermo"
          >
            Palermo
          </option>
          <option
            value="La Spezia (Cinque Terre)"
          >
            La Spezia (Cinque Terre)
          </option>
          <option
            value="San Francisco"
          >
            San Francisco
          </option>
          <option
            value="New York"
          >
            New York
          </option>
          <option
            value="Honolulu"
          >
            Honolulu
          </option>
          <option
            value="San Diego"
          >
            San Diego
          </option>
          <option
            value="Abu Dhabi"
          >
            Abu Dhabi
          </option>
          <option
            value="New Orleans"
          >
            New Orleans
          </option>
          <option
            value="Cape Town"
          >
            Cape Town
          </option>
          <option
            value="Jeddah"
          >
            Jeddah
          </option>
          <option
            value="Rotterdam"
          >
            Rotterdam
          </option>
          <option
            value="Tenerife"
          >
            Tenerife
          </option>
          <option
            value="Civitavecchia (Rome)"
          >
            Civitavecchia (Rome)
          </option>
          <option
            value="Mosquera Islet"
          >
            Mosquera Islet
          </option>
          <option
            value="Dubai"
          >
            Dubai
          </option>
          <option
            value="Tokyo"
          >
            Tokyo
          </option>
          <option
            value="Palma de Mallorca"
          >
            Palma de Mallorca
          </option>
          <option
            value="Southampton"
          >
            Southampton
          </option>
        </select>
      </div>
    </div>
    <div
      class="filter"
    >
      <label
        class="label"
      >
        Cruiseline
      </label>
      <div
        class="select wide naked"
      >
        <select
          class="inner"
          data-testid="select"
        >
          <option
            value="any"
          >
            Any
          </option>
          <option
            value="MSC Cruises"
          >
            MSC Cruises
          </option>
          <option
            value="Holland America Line"
          >
            Holland America Line
          </option>
          <option
            value="Princess Cruises"
          >
            Princess Cruises
          </option>
          <option
            value="Royal Caribbean International"
          >
            Royal Caribbean International
          </option>
          <option
            value="Norwegian Cruise Line"
          >
            Norwegian Cruise Line
          </option>
          <option
            value="Seabourn Cruise Line"
          >
            Seabourn Cruise Line
          </option>
          <option
            value="Carnival Cruise Line"
          >
            Carnival Cruise Line
          </option>
          <option
            value="Celebrity Cruises"
          >
            Celebrity Cruises
          </option>
          <option
            value="Oceania Cruises"
          >
            Oceania Cruises
          </option>
          <option
            value="Azamara"
          >
            Azamara
          </option>
          <option
            value="Cunard"
          >
            Cunard
          </option>
        </select>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`app/components/molecules/SailingFilters stories SelectedPort matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="filters"
  >
    <div
      class="filter"
    >
      <label
        class="label"
      >
        Departure port
      </label>
      <div
        class="select wide naked"
      >
        <select
          class="inner"
          data-testid="select"
        >
          <option
            value="any"
          >
            Any
          </option>
          <option
            value="Naples"
          >
            Naples
          </option>
          <option
            value="Sydney"
          >
            Sydney
          </option>
          <option
            value="Brisbane"
          >
            Brisbane
          </option>
          <option
            value="Singapore"
          >
            Singapore
          </option>
          <option
            value="Le Havre (Paris)"
          >
            Le Havre (Paris)
          </option>
          <option
            value="Colon"
          >
            Colon
          </option>
          <option
            value="Barcelona"
          >
            Barcelona
          </option>
          <option
            value="Port Canaveral"
          >
            Port Canaveral
          </option>
          <option
            value="Los Angeles"
          >
            Los Angeles
          </option>
          <option
            value="Fort Lauderdale (Port Everglades)"
          >
            Fort Lauderdale (Port Everglades)
          </option>
          <option
            value="Sockhna"
          >
            Sockhna
          </option>
          <option
            value="Lisbon"
          >
            Lisbon
          </option>
          <option
            value="Miami"
          >
            Miami
          </option>
          <option
            value="Isla Baltra"
          >
            Isla Baltra
          </option>
          <option
            value="Genoa"
          >
            Genoa
          </option>
          <option
            value="Hamburg"
          >
            Hamburg
          </option>
          <option
            value="Palermo"
          >
            Palermo
          </option>
          <option
            value="La Spezia (Cinque Terre)"
          >
            La Spezia (Cinque Terre)
          </option>
          <option
            value="San Francisco"
          >
            San Francisco
          </option>
          <option
            value="New York"
          >
            New York
          </option>
          <option
            value="Honolulu"
          >
            Honolulu
          </option>
          <option
            value="San Diego"
          >
            San Diego
          </option>
          <option
            value="Abu Dhabi"
          >
            Abu Dhabi
          </option>
          <option
            value="New Orleans"
          >
            New Orleans
          </option>
          <option
            value="Cape Town"
          >
            Cape Town
          </option>
          <option
            value="Jeddah"
          >
            Jeddah
          </option>
          <option
            value="Rotterdam"
          >
            Rotterdam
          </option>
          <option
            value="Tenerife"
          >
            Tenerife
          </option>
          <option
            value="Civitavecchia (Rome)"
          >
            Civitavecchia (Rome)
          </option>
          <option
            value="Mosquera Islet"
          >
            Mosquera Islet
          </option>
          <option
            value="Dubai"
          >
            Dubai
          </option>
          <option
            value="Tokyo"
          >
            Tokyo
          </option>
          <option
            value="Palma de Mallorca"
          >
            Palma de Mallorca
          </option>
          <option
            value="Southampton"
          >
            Southampton
          </option>
        </select>
      </div>
    </div>
    <div
      class="filter"
    >
      <label
        class="label"
      >
        Cruiseline
      </label>
      <div
        class="select wide naked"
      >
        <select
          class="inner"
          data-testid="select"
        >
          <option
            value="any"
          >
            Any
          </option>
          <option
            value="MSC Cruises"
          >
            MSC Cruises
          </option>
          <option
            value="Holland America Line"
          >
            Holland America Line
          </option>
          <option
            value="Princess Cruises"
          >
            Princess Cruises
          </option>
          <option
            value="Royal Caribbean International"
          >
            Royal Caribbean International
          </option>
          <option
            value="Norwegian Cruise Line"
          >
            Norwegian Cruise Line
          </option>
          <option
            value="Seabourn Cruise Line"
          >
            Seabourn Cruise Line
          </option>
          <option
            value="Carnival Cruise Line"
          >
            Carnival Cruise Line
          </option>
          <option
            value="Celebrity Cruises"
          >
            Celebrity Cruises
          </option>
          <option
            value="Oceania Cruises"
          >
            Oceania Cruises
          </option>
          <option
            value="Azamara"
          >
            Azamara
          </option>
          <option
            value="Cunard"
          >
            Cunard
          </option>
        </select>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
